dev/com.ibm.ws.jdbc_fat_driver/test-applications/jdbcapp/src/jdbc/fat/driver/web/JDBCDriverManagerServlet.java

```diff
@@ -401,4 +407,29 @@ public void testGetSetLoginTimeout() throws Exception {
         assertEquals("Login timeout should be 0 regardless of what is done to DriverManager", 0, fatDriverDS.getLoginTimeout());
     }
 
+    //Test that you are unable to set the logwriter on the DataSource and that the getLogWriter method returns null
+    @Test
+    public void testGetSetLogWriter() throws Exception {
+        try {
+            fatDriverDS.setLogWriter(new PrintWriter(System.out));
+        } catch (SQLFeatureNotSupportedException ex) {
+        } //expected
+        assertNull("The getLogWriter method should always return null when using Driver", fatDriverDS.getLogWriter());
+    }
+
+    //Test that ensures unwrapping a DataSource backed by Driver to the underlying Driver interface is not possible
+    @Test
+    @ExpectedFFDC({ "java.sql.SQLException" })
+    public void testUnwrapDriver() throws Exception {
+        try {
+            fatDriverDS.unwrap(Driver.class);
+            fail("Should not be able to unwrap to the Driver interface");
+        } catch (SQLException ex) {
+            assertTrue("Expected exception message to contian DSRA9122E. Message was " + ex.getMessage(), ex.getMessage().contains("DSRA9122E"));
```

Comment:
In order to preserve the full stack for debug purposes, it is better to simply rethrow caught exceptions than to run an assert that only prints the message.

Reply:
We still have the FFDC with that information, but I'm fine changing the test as well